sync/src/chain.rs

```diff
@@ -843,8 +843,8 @@ impl ChainSync {
 			self.downloading_bodies.remove(&n);
 			self.downloading_headers.remove(&n);
 		}
-		self.headers.remove_tail(&start);
-		self.bodies.remove_tail(&start);
+		self.headers.remove_from(&start);
```

Comment:
All following blocks/headers must be removed since it is possible that on a forked chain a block body (same transaction/uncles) will belong to a block with a different number or hash, which will cause a block body-to header mismatch.